pkg/kine/drivers/generic/generic.go

```diff
@@ -315,8 +320,16 @@ func (d *Generic) query(ctx context.Context, txName, sql string, args ...interfa
 		if err != nil {
 			err = fmt.Errorf("query (try: %d): %w", i, err)
 		}
+		d.ac.Metrics.NumCurrentTxn.Add(-1)
 		recordOpResult(txName, err, start)
 	}()
+
+	d.ac.Metrics.NumCurrentTxn.Add(1)
```

Comment:
I'm not really happy about how this is implemented. I'd rather have the admission control only once but could not really figure out a unique entry point for all requests where this could be done (without refactoring large portions of the code.

Happy to hear opinions/alternative suggestions.